File: src/ch2/cmptext.py
# ファイル名を指定 --- (*1)
afile = "./fizzbuzz_python.txt"
bfile = "./fizzbuzz_rust.txt"

# ファイルを文字列として読み込む --- (*2)
with open(afile, "r") as fp:
    astr = fp.read()
with open(bfile, "r") as fp:
    bstr = fp.read()

# 念のため余分な空行をトリム
astr = astr.strip()
bstr = bstr.strip()

# 比較 --- (*3)
if astr == bstr:
    print("ok")
else:
    print("ng")

File: src/ch2/cmptext.rs
// ファイル操作のライブラリを読む --- (*1)
use std::fs;
fn main() {
    // ファイル名を指定 --- (*2)
    let afile = "./fizzbuzz_python.txt";
    let bfile = "./fizzbuzz_rust.txt";
    
    // ファイルを文字列として読む --- (*3)
    let astr = fs::read_to_string(afile).unwrap();
    let bstr = fs::read_to_string(bfile).unwrap();
    
    // 念のためトリム
    let astr = astr.trim();
    let bstr = bstr.trim();

    // 比較 --- (*4)
    if astr == bstr {
        println!("ok");
    } else {
        println!("ng");
    }
}

